storage/innobase/dict/dict0dict.cc

```diff
@@ -5841,7 +5841,7 @@ an earlier upgrade. This will update the table_id by adding DICT_MAX_DD_TABLES
 void dict_table_change_id_sys_tables() {
   ut_ad(dict_sys_mutex_own());
 
-  for (uint32_t i = 0; i < SYS_NUM_SYSTEM_TABLES; i++) {
+  for (int i = SYS_NUM_SYSTEM_TABLES - 1; i >= 0; i--) {
```

Comment:
Can you please check if the problem exists for user tables?
Also, please add a zipped 5.7 datadir and testcase

Reply:
problem not exists for user tables. as they are loaded after mapping. added the test case

Reply:
right, another reason could be that every user table is loaded +1024 offset. So there is no conflict.
Incase of SYS_tables, all are loaded with existing ids. Then we change ids of SYS tables.